ERC20 support

rahulkhairnarr opened the pull request.

yivo committed:
Handle missing Redis values in GET /api/v2/k (fixes #1281) (#1295)

yivo committed:
Fix Figaro warnings (#1300)

yivo committed:
Fix Google auth error (Error: invalid_request) which breaks local sign in for development (#1301)

yivo committed:
Prevent race conditions in withdraw worker + add rich logging. (#1306)

yivo committed:
Handle race conditions when registering member + add useful logging for OmniAuth sequence. (#1310)

yivo committed:
Handle missing member & currency as validation errors preventing NoMethodError (closes #1272) (#1299)

kite-bot committed:
[ci skip] Bump 1.8.37.

dinesh-skyach committed:
Improvements for legacy benchmark tools (closes #1249) (#1254)

* Improvements for legacy benchmark tools

* Moved profiling under benchmark and reduced rounds for benckmark tasks

kite-bot committed:
[ci skip] Bump 1.8.38.

yivo committed:
Add GET /api/v2/member_levels (#1321)

kite-bot committed:
[ci skip] Bump 1.8.39.

dinesh-skyach committed:
Matching engine and trade executor generating errors (#1328)

* Matching engine and trade executor generating errors

kite-bot committed:
[ci skip] Bump 1.8.40.

dinesh-skyach committed:
Fix startup problems in benchmark (fixes #1329) (#1344)

* Fixed benchmark tools

kite-bot committed:
[ci skip] Bump 1.8.41.

Commit:
Adding docker compose files for backend services (#1354)

vshatravenko committed:
Added ability to configure production db name from env (#1355)

kite-bot committed:
[ci skip] Bump 1.8.42.

ritesh-skyach committed:
Add missing error messages in APIv2 (closes #1333) (#1343)

* Fix error message in API endpoint

* Added unverfied member specs

* Added trade unverified member spec

kite-bot committed:
[ci skip] Bump 1.8.43.

ritesh-skyach committed:
Fix XRP destination tag bug which breaks XRP withdraws (closes #1311) (#1332)

* Fix XRP withdraw seems broken

* Some fixes

* Missing argument

dinesh-skyach committed:
Specs for WS protocol API (#1322)

* Specs for WS protocol API

* updates

kite-bot committed:
[ci skip] Bump 1.8.44.

vitorgamer58 commented:
eu não sei se simplesmente você fazer um pull request do fork deles já tornaria seu fork compatível com tokens ERC20, é melhor usar logo o fork deles, estou tentando adicionar tokens ERC20 em minha exchange também.

dinesh-skyach committed:
Added Invalid not supported market error (#1370)

kite-bot committed:
[ci skip] Bump 1.8.45.

yivo committed:
Optimizations for trade executor (#1335)

* Optimizations for trade executor

kite-bot committed:
[ci skip] Bump 1.8.46.

yivo committed:
Document every daemon (#1377)

Commit:
Starting development v1.9.0-alpha

yivo committed:
Update sprockets gem (#1386)

Commit:
Adding geth backend

yivo committed:
Remove solvency feature (#1390)

yivo committed:
Update ffi to 1.9.25 (#1393)

dinesh-skyach committed:
Ability to get data between some time interval (time_from, time_to) in GET /api/v2/k (closes #1290) (#1342)

* get data in specific time interval

k1T4eR committed:
Send label when generating BitGo address. (#1318)

yivo committed:
Allow users to have multiple deposit addresses (#1282)

dinesh-skyach committed:
Added 24 hours currency trades API endpoint (#1368)

dinesh-skyach committed:
Fixed Broken market spec (#1407)

* Fixed broken market spec

yivo committed:
Remove auditing system (you have to use Event API to do audits now) (#1391)

ritesh-skyach committed:
Withdrawals show transaction id (#1411)

* Withdrawals show transaction id

* Convert txid to link

ritesh-skyach committed:
Change Default Domain To peatio.tech (#1450)

Commit:
Add API endpoint for currencies (#1433)

* Add API endpoint for currencies

* Prettify code for currency/trades endpoint

* Prettify tests for currency/trades endpoint

* Finish

* Fix spec

* Add one more spec of currency structure

* Fix typo

* Add documentation about added currency API

* Fix

ysv committed:
Include PublishToRabbitMQ GenerateJWT Event API middlewares by default (closes #1457) (#1458)

* Include PublishToRabbitMQ GenerateJWT Event API middlewares by default

tamadamas committed:
Event API serializers imporvements (closes #1376, #1396) (#1442)

* #1376: Add account model event
* #1396: Add id to order and trade events

Commit:
Fix migration multiple_deposit_addresses (#1402)

Atul9 committed:
Fix typo (#1496)

Thanks @Atul9

Commit:
Add more details for the API docs (#1493)

* Add security definitions and contact info for swagger spec
* Add more documentation and swagger stuff for currencies endpoint

Commit:
Fix typo in setup-osx.md documentation (#1492)

Commit:
Add ability to specity min price (closes #1447) (#1460)

Commit:
Adding Peatio Core

Commit:
Add icon url to currency (closes #1449) (#1465)

* Add icon url to currency (closes #1449)

* Add icon_url in api currency entities. Minor related refactoring

Commit:
Updating patch level for gems (#1512)

* Updating patch level for gems

* Updating mini_racer

* Changing sass to sassc

* Fixed middleware specs failure due to gem update (#1513)

Commit:
Integrate Blockchain and Wallet model and services with new transaction processing and multi wallet support  (#1404)

* Blockchain model and Database seeding

* Adding wallet seeding

* Adding wallet model

* Fix STI problem

* Add missing associations

* Fix indexes order

* Adding factories and a test stub

* BlockchainService #process_blockchain deposits with proof of work (#1417)

* Added blockchain service

* Ability to register a blockchain/wallet from Admin Panel (#1422)

* Revert some changes (related to #1422)

* Single BlockchainService per Blockchain (#1424)

* Added Wallet/Blockchain validations (#1429)

* Added Wallet/Blockchain validations

* Confirm withdrawals in BlockchainService #process_blockchain (#1427)

* Withdraw confirm new behavior

* Refactor Services

* Build transaction refactor

* Fix wallets seed & update blockchain daemon

* Use min_confirmations instead of withdraw & deposit

* Get rid of deposit_confirmations

* Withdraw process to confirming

* Rebase on master

* Bitcoin Blockchain Service (#1444)

* Improve BlockchainService logger. Wallet & Blockchain bugfixes (#1474)

* Add Blockchain Key In Currency (#1473)

*  Remove CoinAPI & daemons. Rename Client to BlockchainClient (#1476)

* Remove CoinAPI & daemons

* Remove CoinAPI & daemons 2

* Rename Client to BlockchainClient

* BlockchainService improve performance

* Fix specs

* Replace Confirmation With Block Number (#1463)

* Add gateway & max_balance to wallets (#1478)

* Wallet per currency

* WalletService module and WalletService::Base class (#1479)

* WalletService module and WalletService::Base class

* WalletClient & WalletClient::Base

* Added deposit coin address daemon (#1480)

* Improvements in Deposit coin address daemon (#1482)

* Geth deposit collectio (eth only)

* Bitcoind deposit collection

* ERC20 deposit_collection

* Withdraw Coin Daemon (#1485)

* Added withdrawal for ETH/ERC20

* Added BTC withdraw via bitcoind

*  DepositCollectionFees worker for ERC20 (#1489)

* DepositCollectionFees worker for ERC20 deposit collection

* Fix Withdraw confirmation bug

* Update doc & daemons.god

* Fix specs

* Litecoin/Dash/BitcoinCash Blockchain Services (#1475)

* Added Litecoin/Dash/BitcoinCash Blockchain Services

* Added Bitcoincash Service/Specs

* rebased with feature/blockchains

* rebased with feature/blockchains

* rebased and updates

* Added wallet client/service for litecoin/bitcoincash/dash

* Updates for admin panel (#1501)

* Use testnet explorer links for testnet blockchains
* Fix coins explorer links removal

* Fix erc20 deposit for tx with empty receipt (#1502)

* Fix wrong client for existing blockchain on admin panel (#1504)

* Bitgo wallet Client/Service (#1491)

* Added Bitgo wallet client/service

* Added dynamic txn estimate fee for deposit_collection daemon

* Fixed withdraw stuck in confirming (#1507)

* Improved dynamic txn fees for bitcoind/bitgo (#1509)

* Feature/blockchains wallets (#1510)

* Confirm withdrawals in BlockchainService #process_blockchain (#1427)

* Withdraw confirm new behavior

* Refactor Services

* Build transaction refactor

* Fix wallets seed & update blockchain daemon

* Use min_confirmations instead of withdraw & deposit

* Get rid of deposit_confirmations

* Withdraw process to confirming

* Admin Panel Modification

* Added ability to disable currencies, markets and wallets

* Fix gateway form

* Wallet Model Changes And Spec Fixes

* Add Missing File

* Wallet Service Changes

* Fixed icons and renamed fields on Market New and Show pages (#1515)

* Upgrading peatio core

Commit:
Updating CHANGELOG for 2018

ritesh-skyach committed:
DesignFix (#1519)

ritesh-skyach committed:
Change Currency Id Limit (#1526)

ysv committed:
Code polish and minor bugfix (closes #1531) (#1529)

* Fix Wallet duplication on each wallets:seed

* Add missing encoding and frozen_string_litteral annotations

* Remove comment

* Refactor

* Refactor html.erb

* Remove unused code from Services and Clients

* WalletService::Base refactor

* Fix specs

* Fix specs 2

* [ci skip] Comment wallets and blockchains setting files

ritesh-skyach committed:
Add Bitgo Wallet ID And Remove Wallet Address In Wallet Model (#1530)

* Add Bitgo Wallet Id In Wallet

* Remove Bitgo Wallet Address

dinesh-skyach committed:
Currencies dead code clean up (#1518)

* Add Bitgo Wallet Id In Wallet

* Remove Bitgo Wallet Address

* Clean up dead code

* Removed case_sensitive from currency
* Added static case_sensitive value to blockchain clients
* Add missing downcase to Deposit and Withdraw
* Validate blockchain before Currency create
* Fix rake seed:wallets

alinetskyi committed:
Edit comments in templates for wallets.yml (#1533)

Commit:
Backport Inspect Address

dinesh-skyach committed:
Fixed rake db:seed failure (#1541)

ysv committed:
Improved updation of blockchain height and update exception handling mechanism (#1534)

ritesh-skyach committed:
Add Wallet Support For Bitcoin Cash Legacy Address (#1540)

Commit:
Spec Changes

dinesh-skyach committed:
Updated xit Specs (#1542)

ritesh-skyach committed:
Do not allow to disable last enabled market (closes #1548) (#1543)

* Do Not Allow To Disable Last Currency And Market

* Fix dependent disable && update currencies seed

dinesh-skyach committed:
Ability to see Transaction ID in user interface (#1546)

* Ability to see Transaction ID in user interface

ysv committed:
[ci skip] Update Readme (#1551)

* [ci skip] Minimalistic local development environment with docker-compose & minor changes

* [ci skip] Update backend.yml

Commit:
Added Backport address validation via wallet client

dinesh-skyach committed:
Fixed wallet secret not being saved from admin panel (#1552)

tamadamas committed:
Add riple blockchain service and client

Commit:
Integrate peatio-core Peatio::Auth::JWTAuthenticator to APIv2 (closes #1555)

mnaichuk committed:
Update docs about authenticating in management api

bpotemskyi committed:
Release Notes for Peatio 1.9 (#1557)

kaiten-mahajan committed:
Return Error if invalid from and to params

Commit:
Informative message when POST /api/v2/orders with wrong volume

dinesh-skyach committed:
Added validation to restrict display currency to be disabled (#1559)

* Added validation to restrict display currency to be disabled

kaiten-mahajan committed:
updates

Commit:
Updating travis credentials (#1572)

Commit:
Add Deposit Coin Address Spec

mnaichuk committed:
Add rc tag for master releases in bump script

kovetskiy committed:
daemons/global_state: use Peatio::MQ::Events instead of Pusher

Commit:
Updating Travis

kovetskiy committed:
upgrade peatio to 0.4.0

Commit:
Add Deposit Collection Spec

Commit:
Remove Features Specs

kaiten-mahajan committed:
Added new test in trades

kovetskiy committed:
replace PusherMarket and PusherMember with Peatio::MQ::Events

kaiten-mahajan committed:
Removed webhook controller and dependencies

Commit:
Support Cash Address format

Commit:
Confirmation can't be negative

Commit:
Spec changes

Commit:
Removing Chromium from Dockerfile

kaiten-mahajan committed:
Updated specs

Commit:
Remove websocket and stuff

mnaichuk committed:
Remove peatio-trading-ui and dependencies from CI

kaiten-mahajan committed:
Extract duplicated code to HasOneBlockchainThroughCurrency (closes #1593)

Commit:
Ability to retrieve address in both formats cash and legacy via API

Commit:
Fix swagger doc for APIv2 (closes #1605)

Commit:
Added currency address format validation

Commit:
updates

Commit:
Human Readable Time Stamps

shal committed:
Update bump script to support rc in tag

Commit:
Add Tools Spec For API

kite-bot committed:
[ci skip] Bump 1.9.0-rc.2.

dinesh-skyach committed:
Fixed broken specs (#1620)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.3.

dinesh-skyach committed:
Added Swagger docs spec (#1612)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.4.

ysv committed:
Execute all block updates in single SQL transaction. Find withdrawals and deposits by txid only (closes #1549, closes #1568) (#1622)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.5.

Commit:
Handle ActiveRecord::RecordNotFound in member api_v2 remove method duplication (closes #1632)

ysv committed:
Move versions to Peatio::Application. Remove lib/peatio (closes #1619) (#1636)

dinesh-skyach committed:
Removed unwanted options in currency model and seed file (closes #1611) (#1618)

ritesh-skyach committed:
Don't open new tabs in Admin Panel & minor refactoring (closes #1609) (#1613)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.6.

tamadamas committed:
Drop spaces from admins list

Commit:
Repository clean up. Minor doc updates (closes #1637, closes #1523)

ysv committed:
Allow to use currencies in both downcase & upcase in Member API (closes #1660) (#1661)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.7.

Commit:
Added explorer_transaction and explorer_address to currency entities (closes #1664)

Commit:
Add docs about performance

Commit:
Add missing API call for getting member accounts (related to #919)

Commit:
Add Bullet for detecting N+1 problems

Commit:
Add grape_strip to resolve whitespace in input params

Commit:
Add missing address in wallets seed config

Commit:
Apigate integration

Integrate Ambassador gateway.
Remove OAuth Authentication and Peatio session support
New grape api organisation into account, market, public endpoints
Remove CORS

Co-authored-by: Yaroslav Savchuk <ysavchuk@heliostech.fr>
Co-authored-by: Maksim Naichuk <mnaichuk@heliostech.fr>
Co-authored-by: Louis Bellet <lbellet@openware.com>

Commit:
Various fixes in docs (#1672)

* Various fixes in docs

* Update README.me

Commit:
Add missing url validations to rails models (#1656)

Closes #1652.

- Add validation for server atttribute in Blockchain model.
- Add validation for url atttribute in Wallet model.

dinesh-skyach committed:
Fetch latest block transactions using getrawmempool for bitcoin-like blockchains (closes #1621) (#1649)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.8.

ymasiuk committed:
Ranger integration (#1686)

* Ranger Integration With Trading-UI
* Add some vars to gon for Ranger
* Delete uneeded variable

kite-bot committed:
[ci skip] Bump 1.9.1-rc.9.

Commit:
Remove all n+1 queries

Commit:
Delete unused daemon and move event_api.md to docs (#1702)

dinesh-skyach committed:
Changed address limits (#1701)

ysv committed:
K line daemon & API fix & specs improvement

* closes #1689
* closes #1693
* closes #1697

rsaphala committed:
rake yarn install now prints the stdout response (#1690)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.10.

kite-bot committed:
[ci skip] Bump 1.9.1-rc.11.

dependabot-support committed:
Bump rack-attack from 5.2.0 to 5.4.1

Bumps [rack-attack](https://github.com/kickstarter/rack-attack) from 5.2.0 to 5.4.1.
- [Release notes](https://github.com/kickstarter/rack-attack/releases)
- [Changelog](https://github.com/kickstarter/rack-attack/blob/master/CHANGELOG.md)
- [Commits](https://github.com/kickstarter/rack-attack/compare/v5.2.0...v5.4.1)

Signed-off-by: dependabot[bot] <support@dependabot.com>

Commit:
Add min deposit amount for currencies and validation for small deposits

Commit:
Updated websocket api doc

dependabot-support committed:
[Security] Bump nokogiri from 1.8.4 to 1.8.5

Bumps [nokogiri](https://github.com/sparklemotion/nokogiri) from 1.8.4 to 1.8.5. **This update includes security fixes.**
- [Release notes](https://github.com/sparklemotion/nokogiri/releases)
- [Changelog](https://github.com/sparklemotion/nokogiri/blob/master/CHANGELOG.md)
- [Commits](https://github.com/sparklemotion/nokogiri/compare/v1.8.4...v1.8.5)

Signed-off-by: dependabot[bot] <support@dependabot.com>

ymasiuk committed:
Add env RANGER_CONNECT_SECURE (#1718)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.12.

Commit:
Fix problem when we publish message but Deposit doesn't exist in db

Commit:
Add new state to deposit model to fix deposit recurrence

Commit:
Implement Fee wallet for deposit collection fees daemon and fix deposit fees recurrence

tamadamas committed:
Remove warning from event api initializer (#1755)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.13.

kite-bot committed:
[ci skip] Bump 1.9.1-rc.14.

Commit:
Update comment and values in currencies.yml. Final code refactor

Commit:
Experimental peatio plugin system v2

Commit:
Add docs/plugins.md

Commit:
[ci skip] Update md

Commit:
Fix Dockerfile

Commit:
[ci skip] MD minor changes

Commit:
Add BULLET to application.yml.erb disable by default

kite-bot committed:
[ci skip] Bump 1.9.1-rc.15.

Commit:
Expose ask_id & order_id in trade event of websocket api

Commit:
Updating Ruby and Gems

Adding gem rubocop-rspec
Correcting factories syntax

Commit:
fix bug on #1775

kite-bot committed:
[ci skip] Bump 1.9.1-rc.16.

kite-bot committed:
[ci skip] Bump 1.9.1-rc.17.

dependabot-support committed:
[Security] Bump loofah from 2.2.2 to 2.2.3

Bumps [loofah](https://github.com/flavorjones/loofah) from 2.2.2 to 2.2.3. **This update includes security fixes.**
- [Release notes](https://github.com/flavorjones/loofah/releases)
- [Changelog](https://github.com/flavorjones/loofah/blob/master/CHANGELOG.md)
- [Commits](https://github.com/flavorjones/loofah/compare/v2.2.2...v2.2.3)

Signed-off-by: dependabot[bot] <support@dependabot.com>

vpetrusenko committed:
Add missing association for fiat

kite-bot committed:
[ci skip] Bump 2.0.1.pre.alpha.0.

kite-bot committed:
[ci skip] Bump 1.9.1-rc.18.

dependabot-support committed:
[Security] Bump rack from 1.6.10 to 1.6.11

Bumps [rack](https://github.com/rack/rack) from 1.6.10 to 1.6.11. **This update includes security fixes.**
- [Release notes](https://github.com/rack/rack/releases)
- [Changelog](https://github.com/rack/rack/blob/master/CHANGELOG.md)
- [Commits](https://github.com/rack/rack/compare/1.6.10...1.6.11)

Signed-off-by: dependabot[bot] <support@dependabot.com>

Commit:
Add validation for order price

kite-bot committed:
[ci skip] Bump 1.9.1-rc.19.

kite-bot committed:
[ci skip] Bump 1.9.1-rc.20.

shal committed:
Compile assets in production mode (fixes #1695)

Commit:
Disable api call for creating new deposit address

kite-bot committed:
[ci skip] Bump 1.9.1-rc.21.

kite-bot committed:
[ci skip] Bump 1.9.1-rc.22.

Commit:
Fix pool deposit issue

kite-bot committed:
[ci skip] Bump 1.9.1-rc.23.

Commit:
Update package-lock.json

kite-bot committed:
[ci skip] Bump 1.9.1-rc.24.

Commit:
Implement skimming deposit collection mechanism based on wallet max_balance (closes #1653) (#1735)

* Added Wallet Service Specs

* Implement skimming deposit collection mechanism for bitcoin

* Implement skimming deposit collection mechanism for eth

* Implement skimming deposit collection mechanism for ripple

* Enumerize Wallet kind

* Edit skiming mechanism for ripple wallet

* Edit skiming mechanism for eth wallet

* Use hash instead array for deposit collections

* Update specs with new skiming mechanism and add specs for ripple

* Add min_collection_amount method to currency

* Move load_balance! request to blockchain service and add specs for wallet service

* Add min_collection_amount column to currency and move spread deposit method to protected

kite-bot committed:
[ci skip] Bump 1.9.1-rc.25.

Commit:
Rewrite quick withdraw limit mechanism, add 24h and 72h withdraw limits (#1817)

kite-bot committed:
[ci skip] Bump 1.9.1-rc.26.

ysv committed:
[ci skip] Bump 2.0.0-alpha. (#1837)

ysv committed:
Rework Peatio accounting using double entry accounting system  (#1777)

  Add Operation Base model and tables

  Add Operations::Asset,Liability,Revenue,Expense
  Add Operations::Chart which define class of items for which money spent or received
  Record Asset, Liability and Revenue operations on Deposit accept
  Record Liability transfer on Order create
  Record Liability transfer on Order cancel
  Record Liability transfer and Revenue creation on Trade create
  Deprecate legacy funds plus lock unlock and subtract funds
  Add Admin rubric for tracking Operations and Balance sheet
  Add Admin with Balance sheet

kite-bot committed:
Store Currency options as JSON and define accessors

Commit:
Income Statement (#1845) (#1849)

* Income Statement (#1845)

Commit:
Release script updated

Commit:
v2.0.1-alpha [skip ci]

Commit:
[ci skip] Adding db schema

ysv committed:
Remove REDIS_PASSWORD env which is always overridden by REDIS_URL (#1856)

Commit:
v2.0.2-alpha [skip ci]

Commit:
Add manualy loading withdrawal from admin panel and validations for events

Commit:
Ability to set custom gas_limit & gas_price per currency.

Commit:
v2.0.3-alpha [skip ci]

Commit:
[ci skip] Updating CHANGELOG.md and api docs

Commit:
Specs for account/balances and code polishing

Commit:
Ability to use null reference for operation

Commit:
[WIP] Operations Management API

Commit:
Improve deposit collection daemon and add ability to reject fiat deposit

Commit:
[WIP] Management::Entities::Operation

Commit:
Entities::Operation & specs & Operation factory

Commit:
WIP

Commit:
v2.0.4-alpha [skip ci]

ysv committed:
Extend entities specs

ysv committed:
List assets,revenues,expenses with specs

Commit:
Add missing method

Commit:
List liability with specs

Commit:
v2.0.5-alpha [skip ci]

Commit:
Managemetn API for creating asset, revenue, expense and libility with specs

Commit:
Update specs & Minore spec refactor

denstepa committed:
use local seeds

*Problem*: workbench has old seeds config files
*Solution*: use local seeds

Commit:
v2.0.6-alpha [skip ci]

chumaknadya committed:
Add opportunity to define minimum amount for trade

Commit:
v2.0.7-alpha [skip ci]

Commit:
v2.0.8-alpha [skip ci]

Commit:
Add min withdraw amount for currencies

Commit:
Add missing deposit "skipped" translation (#1926)

Commit:
Load balance from hot wallet before make quick withdraw (#1935)

Commit:
v2.0.9-alpha [skip ci]

Commit:
Remove ActionDispatch Session. Disable CSRF

Commit:
Add Currency Name

Commit:
v2.0.10-alpha [skip ci]

Commit:
v2.0.11-alpha [skip ci]

Commit:
Add automatic docs generation for v2

Commit:
Account Withdraw API With OTP

* Add POST /api/v2/account/withdraws
* Integrate Vault to travis & compose
* Integrate Vault::TOTP client
* Add ENABLE_ACCOUNT_WITHDRAWAL_API, VAULT_URL & VAULT_TOKEN envs

Co-authored-by: Yaroslav Savchuk <ysavchuk@heliostech.fr>
Co-authored-by: Dinesh Chohda <dinesh.skyach@gmail.com>

Commit:
v2.0.12-alpha [skip ci]

Commit:
Change http status for permitting actions

Commit:
Public version API endpoint

Commit:
Add CONTRIBUTING guide [skip ci]

Commit:
v2.0.13-alpha [skip ci]

Commit:
Merge pull request #1966 from mnaichuk/2-0-add-cors

Add an ability to specify CORS headers

Commit:
Rewrite api entities for docs. Split user and management api (#1958)

Commit:
Add swagger.json in prettyjson format (#1984)

Commit:
v2.0.14-alpha [skip ci]

Commit:
Expose Trade market maker and don't expose null side

* closes #1929
* closes #1920

Commit:
v2.0.15-alpha [skip ci]

Commit:
Add Peatio UI removal alert

ysv committed:
Publish WS event on k-line updates. Improve k-daemon performance

* Publish MQ event on  #update_point or #append_point
* Update WS API doc
* Skip unnecessary k point updates
* Use lset instead of pop & push
* Add TODO to k.rb
* Add k-line comments
* KLineService #humanize_period(period)

closes #1995

Commit:
v2.0.16-alpha [skip ci]

Commit:
Merge pull request #1989 from mnaichuk/new-market-api

Improve markets and currencies API

Commit:
Rewrite get_depth to fetch data from db (#1894)

Commit:
Improve deposit collection fees daemon (#1925)

ysv committed:
Add close, avg_price & price_change_percent to WS & REST API (#2000)

* Add close, avg_price & price_change_percent
* Update WS API doc
* Move & refactor Market #change_ratio to Global
* Add ticker specs

closes #1991

Commit:
Fix private order update event (#2007)

Commit:
Fix markets seed file (#2009)

Commit:
v2.0.17-alpha [skip ci]
